File: Java/Algorithms/Sorting/InsertionSort.java
//Insertion Sort it search in all items less than i
//Take O(n*n)

package Learn;


public class InsertionSort {
    
    private void Swap(int array[] , int x , int y){
        int temp = array[x];
        array[x] = array[y];
        array[y] = temp;
    }


    public void Sorting(int array[]){
        for(int i = 1 ; i < array.length ; i++){
            for(int a = i ; a > 0 ; a--){
                if(array[a] < array[a - 1]){
                    Swap(array,a,a-1);
                } 
            }
        }  
    }
    
}

File: Python/Algorithms/Sorting/InsertionSort.py
#Insertion Sort
def sorting(array):
    for i in range(1,len(array),1):
        for a in range(i,0,-1):
            if(array[a] < array[a-1]):
                array[a] , array[a-1] = array[a-1] , array[a]
